tests/cecil-tests/Documentation.cs

```diff
@@ -0,0 +1,229 @@
+						foreach (var failure in fixedFailures.Take (10))
+							Console.WriteLine ($"    {failure}");
+						if (fixedFailures.Count > 10)
+							Console.WriteLine ($"    ... and {fixedFailures.Count () - 10} more");
+					}
+					Assert.Fail ($"Found {unknownFailures.Count} undocumented APIs (not known failures) and {fixedFailures.Count} APIs that were marked as known failures but are now documented. If this is expected, set the WRITE_KNOWN_FAILURES=1 environment variable, run the test again, and commit the modified known failures file.");
+				}
+			}
+		}
+
+		static HashSet<AssemblyApi> GetDocumentedMembers (string xml)
+		{
+			var rv = new HashSet<AssemblyApi> ();
+			var doc = new XmlDocument ();
+			doc.LoadWithoutNetworkAccess (xml);
+			foreach (XmlNode node in doc.SelectNodes ("/doc/members/member")!) {
+				rv.Add (new AssemblyApi (node.Attributes! ["name"]!.Value!, null));
+			}
+			return rv;
+		}
+
+		class AssemblyApi {
+			public string DocId;
+			public bool? PubliclyVisible;
+
+			public AssemblyApi (string docId, bool? visible)
+			{
+				DocId = docId;
+				PubliclyVisible = visible;
+			}
+
+			public override int GetHashCode ()
+			{
+				return DocId.GetHashCode ();
+			}
+
+			public override bool Equals (object? other)
+			{
+				return other is AssemblyApi aa && aa.DocId == DocId;
+			}
+		}
+
+		static HashSet<AssemblyApi> GetAssemblyMembers (AssemblyDefinition asm)
+		{
+			var rv = new HashSet<AssemblyApi> ();
+
+			foreach (var member in asm.EnumerateMembers ()) {
+				string name;
+
+				if (member is MethodDefinition md) {
+					name = "M:" + GetDocId (md);
+				} else if (member is PropertyDefinition pd) {
+					name = "P:" + GetDocId (pd);
+				} else if (member is FieldDefinition fd) {
+					name = "F:" + GetDocId (fd.DeclaringType) + "." + fd.Name;
+				} else if (member is EventDefinition ed) {
+					name = "E:" + GetDocId (ed);
+				} else if (member is TypeDefinition td) {
+					name = "T:" + GetDocId (td);
+				} else {
+					throw new NotImplementedException ($"Unknown member type: {member.GetType ()}");
+				}
+				rv.Add (new AssemblyApi (name, member.IsPubliclyVisible ()));
+			}
+
+			return rv;
+		}
+
+		// https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/#id-strings
+
+		static string GetDocId (MethodDefinition md)
+		{
+			var methodName = md.Name.Replace ('.', '#');
+			var name = GetDocId (md.DeclaringType) + "." + methodName;
+			if (md.HasGenericParameters)
+				name += $"``{md.GenericParameters.Count}";
+			if (md.HasParameters) {
+				name += "(" + string.Join (",", md.Parameters.Select (p => GetDocId (p.ParameterType))) + ")";
+			}
+
+			if (md.Name == "op_Explicit" || md.Name == "op_Implicit") {
+				name += "~" + GetDocId (md.ReturnType);
+			}
+
+			return name;
+		}
+
+		static string GetDocId (EventDefinition ed)
+		{
+			return GetDocId (ed.DeclaringType) + "." + ed.Name;
+		}
```

Comment:
nit slightly cleaner syntax

```suggestion
		static string GetDocId (EventDefinition ed) => GetDocId (ed.DeclaringType) + "." + ed.Name;
```